code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/WebViewGestureListener.java

```diff
@@ -200,16 +200,14 @@ public void onAnimationRepeat(final Animator animator) {}
     private void dismissMessage(final MessageGesture gesture, final boolean dismissedWithGesture) {
         parentFragment.dismissedWithGesture = dismissedWithGesture;
 
-        if (parentFragment.message != null
-                && parentFragment.message.fullScreenMessageDelegate != null) {
+        if (parentFragment.message != null && parentFragment.message.listener != null) {
             final String behavior =
                     parentFragment.gestures == null ? null : parentFragment.gestures.get(gesture);
 
             // if we have a gesture mapping with behaviors, use the specified behavior. otherwise,
             // just dismiss the message.
             if (!StringUtils.isNullOrEmpty(behavior)) {
-                parentFragment.message.fullScreenMessageDelegate.overrideUrlLoad(
-                        parentFragment.message, behavior);
+                parentFragment.message.listener.overrideUrlLoad(parentFragment.message, behavior);
```

Comment:
Do you mean to call the one from `MessagingDelegate`? Why are you not using the return value?

Reply:
the fullscreen message delegate is called as the gesture map has a mapping between swipes and interaction tracking:
```
\"gestures\":{\"swipeUp\":\"adbinapp://dismiss?interaction=swipeUp\",\"swipeDown\":\"adbinapp://dismiss?interaction=swipeDown\",\"swipeLeft\":\"adbinapp://dismiss?interaction=swipeLeft\",\"swipeRight\":\"adbinapp://dismiss?interaction=swipeRight\",\"tapBackground\":\"adbinapp://dismiss?interaction=tapBackground\"}
```
the overrideUrlLoad function will handle the interaction tracking based on the type of behavior/gesture that occurred to dismiss the message

Reply:
the return value is not used because the url being passed is one to be handled by the messaging extension only.